Framework/PythonInterface/plugins/algorithms/FitIncidentSpectrum.py

```diff
@@ -0,0 +1,221 @@
+            parsed_fit = self.parse_binning(binning_for_fit)
+            if parsed_fit[0] < range_min - bin_width:
+                issues['BinningForFit'] = "Lower bin range outside of InputWorkspace range, minimum value: %f" % (
+                    range_min - bin_width)
+            if parsed_fit[-1] > range_max + bin_width:
+                issues['BinningForFit'] = "Upper bin range outside of InputWorkspace range, maximum values: %f" % (
+                    range_max + bin_width)
+            if parsed_calc.size == 0:
+                issues['BinningForFit'] = "Invalid parameters for bin range"
+        return issues
+
+    def _setup(self):
+        self._input_ws = self.getProperty('InputWorkspace').value
+        self._output_ws = self.getProperty('OutputWorkspace').valueAsStr
+        self._binning_for_fit = self.getProperty('BinningForFit').value
+        self._fit_spectrum_with = self.getProperty('FitSpectrumWith').value
+        self._binning_for_calc = self.getProperty('BinningForCalc').value
+        if self._binning_for_calc is "":
+            x = self._input_ws.readX(0)
+            binning = [str(i) for i in [min(x), x[1] - x[0], max(x) + x[1] - x[0]]]
+            self._binning_for_calc = ",".join(binning)
+        if self._binning_for_fit == "":
+            self._binning_for_fit = self._binning_for_calc
+
+    def PyExec(self):
+        self._setup()
+        x = self.parse_binning(self._binning_for_calc)
+        rebinned = Rebin(
+            self._input_ws,
+            Params=self._binning_for_fit,
+            PreserveEvents=True,
+            StoreInADS=False)
+        incident_index = 0
+        x_fit = np.array(rebinned.readX(incident_index))
+        y_fit = np.array(rebinned.readY(incident_index))
+
+        if len(x_fit) != len(y_fit):
+            x_fit = x_fit[:-1]
+
+        if self._fit_spectrum_with == 'CubicSpline':
+            # Fit using cubic spline
+            fit, fit_prime = self.fit_cubic_spline(x_fit, y_fit, x, s=1e7)
+        elif self._fit_spectrum_with == 'CubicSplineViaMantid':
+            # Fit using cubic spline via Mantid
+            fit, fit_prime = self.fit_cubic_spline_via_mantid_spline_smoothing(
+                self._input_ws,
+                ParamsInput=self._binning_for_fit,
+                ParamsOutput=self._binning_for_calc,
+                Error=0.0001,
+                MaxNumberOfBreaks=0)
+        elif self._fit_spectrum_with == 'GaussConvCubicSpline':
+            # Fit using Gauss conv cubic spline
+            fit, fit_prime = self.fit_cubic_spline_with_gauss_conv(x_fit, y_fit, x, sigma=0.5)
+
+        # Create output workspace
+        output_workspace = CreateWorkspace(
+            DataX=x,
+            DataY=np.append(fit, fit_prime),
+            UnitX='Wavelength',
+            NSpec=2,
+            Distribution=False,
+            StoreInADS=False)
+        self.setProperty("OutputWorkspace", output_workspace)
+
+    def parse_binning(self, binning):
```

Comment:
If we use the `FloatArrayProperty` then the `.value` from `getProperty` should already be a `numpy` array.